kubernetes/README.adoc

```diff
@@ -0,0 +1,78 @@
+= Kubernetes and Openshift tutorial
+
+This tutorial contains instructions on how to run Infinispan library mode (as a microservice) in Kubernetes/OpenShift.
+
+Prerequisites: Maven and Docker daemon running in the background.
+
+== Setting up OpenShift
+
+OpenShift Origin has a command line client called `oc` (it can be downloaded https://github.com/openshift/origin/releases[here]).
+
+The client allows you to spin up a local cluster (of course this is not recommended for production but for testing, it's perfect). In order to do that you need to invoke:
+
+```
+oc cluster up
+```
+
+== Setting up Kubernetes
+
+It is also possible to run this tutorial on Kubernetes. Installing a local cluster requires downloading and installing Minikube (detailed instruction might be found http://kubernetes.io/docs/getting-started-guides/minikube/[here]).
+
+Unfortunately Kubernetes installation does not use local Docker registry. The easiest way to overcome this problem is to push locally built tutorial into docker.io repository. The easiest way to do it is to use fabric8 plugin:
+
+```
+docker login
+.. Log into docker.io using your credentials ..
+
+.. Modify image name in pom.xml (//image/name in fabric8 plugin configuration) ..
+.. Use your own name e.g. slaskawi/infinispan-simple-tutorial ..
+
+mvn install fabric8:push
+```
+
+After performing this step, just invoke:
+
+```
+mvn fabric8:run
+```
+
+== Configuring Service Account for KUBE_PING
+
+Since https://github.com/jgroups-extras/jgroups-kubernetes[KUBE_PING] Discovery Protocol requires viewing all Pods in the OpenShift Project, you need to add additional privileges to the OpenShift:
+
+```
+oc login -u system:admin
+oc policy add-role-to-user view system:serviceaccount:myproject:default -n myproject
+```
+
+== Switch to a normal user
+
+After the OpenShift is running, you need to login as a `developer`. That's the standard role you should always be using:
+
+```
+oc login -u developer -p developer
+```
+
+== Building this tutorial
+
+This tutorial is built using maven command:
+
+```
+mvn install
+```
+
+Note that `target/` directory contains additional directories like `docker` (with generated Dockerfile) and `classes/META-INF/fabric8` with Kubernetes and OpenShift deployment templates.
+
+== Deploying this tutorial to the Kubernetes/OpenShift
+
+This is handles automatically by Fabric8 maven plugin, just invoke:
+
+```
+mvn fabric8:run
+```
+
+== Viewing and scaling out
+
+Everything should be up and running at this point. Now login into the OpenShift web console and scale the application our or in.
+
```

Comment:
When I tried the tutorial, after the containers were up on k8s, I went to the console and could not find a way to scale it.  Scaling via command line required lots of digging to find out the exact command, so IMO it wouldn't hurt to spare the user the same work.